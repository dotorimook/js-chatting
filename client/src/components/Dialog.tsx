import React, { FC } from 'react';
import IComponentProps from 'interfaces/IComponentProps';
import { DialogFrame } from './Frame';
import Button from './form/Button';
import Icon from './Icon';

interface IDialogProps extends IComponentProps {
  visible: boolean;
  onClose: ()=>void;
}

const Dialog:FC<IDialogProps> = (props:IDialogProps) => {
  return (
    <DialogFrame className={props.visible? 'visible':'hidden'}>
      <div className='dialog-frame'>
        <Button className='btn-close'
          onClick={props.onClose}
        >
          <Icon name='close'/>
        </Button>
        {props.children}
      </div>
    </DialogFrame>
  )
}

export default Dialog;