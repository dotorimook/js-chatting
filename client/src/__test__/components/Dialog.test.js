import React from 'react';
import 'setupTest';
import Dialog from 'components/Dialog';
import {shallow} from 'enzyme';

describe('Dialog Test', () => {
  let ui = null;
  const children = <div className='test'></div>;
  const onClose = jest.fn();

  beforeAll(()=>{
    ui = shallow(<Dialog onClose={onClose}>{children}</Dialog>);
  });

  it('matches snapshot', () => {
    expect(ui).toMatchSnapshot();
  });

  it('properly renders chidlren', () => {
    expect(ui.find('div.test')).toHaveLength(1);
  });

  it('has a close button', () => {
    expect(ui.find('.btn-close')).toHaveLength(1);
  });

  it('onClose called when a close button clicked', () => {
    const closeButton = ui.find('.btn-close');
    closeButton.simulate('click');
    expect(onClose).toHaveBeenCalled();
  });
});