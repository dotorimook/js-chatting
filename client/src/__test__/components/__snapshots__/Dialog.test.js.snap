// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog Test matches snapshot 1`] = `
<styled.div
  className="hidden"
>
  <div
    className="dialog-frame"
  >
    <Button
      className="btn-close"
      onClick={[MockFunction]}
    >
      <Icon
        name="close"
      />
    </Button>
    <div
      className="test"
    />
  </div>
</styled.div>
`;
